pkg/types/openstack/validation/platform.go

```diff
@@ -21,6 +22,22 @@ func ValidatePlatform(p *openstack.Platform, n *types.Networking, fldPath *field
 	} else if !isValidValue(p.Cloud, validClouds) {
 		allErrs = append(allErrs, field.NotSupported(fldPath.Child("cloud"), p.Cloud, validClouds))
 	} else {
+		if p.MachinesSubnet != "" {
+			if len(p.ExternalDNS) > 0 {
+				allErrs = append(allErrs, field.InternalError(fldPath.Child("machinesSubnet"), errors.New("externalDNS cannot be set on custom subnets")))
+			}
+			if !validUUIDv4(p.MachinesSubnet) {
+				allErrs = append(allErrs, field.InternalError(fldPath.Child("machinesSubnet"), errors.New("invalid subnet ID")))
+			} else {
+				cidr, err := fetcher.GetSubnetCIDR(p.Cloud, p.MachinesSubnet)
+				if err != nil {
+					allErrs = append(allErrs, field.InternalError(fldPath.Child("machinesSubnet"), fmt.Errorf("invalid subnet %v", err)))
+				}
+				if n.MachineNetwork[0].CIDR.String() != cidr {
+					allErrs = append(allErrs, field.InternalError(fldPath.Child("machinesSubnet"), fmt.Errorf("the first CIDR in machineNetwork, %s, doesn't match the CIDR of the machineSubnet, %s", n.MachineNetwork[0].CIDR.String(), cidr)))
```

Comment:
There is a possibility that there are more than 1 manchine network... Custom network subnet cidr should match at least one cidr from the list, not just the first one.

Reply:
the problem is that we set the VIPs from the first one, So if the cidr doesnt match the first one, then the VIPs will be set to an IP address that is not in the machinesSubnet CIDR range